http://bugs.gentoo.org/280984
Fix byte-compilation of EIEIO with Emacs 23.1.50 (as of 2009-07-23).
Patch from upstream CVS.

--- cedet/eieio/eieio-comp.el	2008/09/29 00:18:57	1.14
+++ cedet/eieio/eieio-comp.el	2009/07/27 11:32:06	1.15
@@ -78,7 +78,10 @@
 	 (class (if (listp arg1) (nth 1 arg1) nil))
 	 (my-outbuffer (if (eval-when-compile
 			     (string-match "XEmacs" emacs-version))
-			   byte-compile-outbuffer outbuffer))
+			   byte-compile-outbuffer 
+			 (condition-case nil
+			     bytecomp-outbuffer
+			   (error outbuffer))))
 	 )
     (let ((name (format "%s::%s" (or class "#<generic>") meth)))
       (if byte-compile-verbose
